schemas/1.0/general/offence.json

```diff
@@ -6,7 +6,7 @@
   "type": "object",
   "properties": {
     "name": { "type": "string" },
-    "offence_class": { "type": ["string", "null"] },
+    "offence_class": { "type": ["string", "null"], "enum": ["A", "K", "G", "B", "I", "J", "D", "C", "H", "F", "E", null] },
```

Comment:
Will the type fail with if the value of the offence is `null` as it is not part of the type as defined here:
https://github.com/ministryofjustice/laa-criminal-legal-aid-schemas/blob/5df43ce58a62d0c4fbbf5ce8521e4967e09c9e30/lib/laa_crime_schemas/types/types.rb#L85

Reply:
No it passes (I had to add null to the end of the enum to get it to work.)

Reply:
What we have to change is `attribute?` to `attribute` (remove the end parenthesis) as we want to enforce the attribute to be present, just not the value that can be null.
But that I believe is part of the bigger piece of work to audit every attribute and ensure we mark them appropriately as mandatory or not, etc. in the Structs, schema, fixtures and everywhere basically 🤣

Reply:
Cool thats fine thanks for explaining 👍